blurring_as_a_service/cleaning_pipeline/source/sampling_pipeline.py

```diff
@@ -0,0 +1,458 @@
+
+    sample_images_for_retraining(grouped_images_by_date: Dict[str, List[str]]) -> None
+        Samples images for retraining based on detection counts and confidence score thresholds, and stores 
+        them in the designated retraining folder.
+
+    collect_images_above_threshold_from_db(grouped_images_by_date: Dict[str, List[str]]) -> pd.DataFrame
+        Collects images with detections above a specified confidence score threshold from the database.
+
+    get_n_random_images_per_date(grouped_images_by_date: Dict[str, List[str]], n_images_to_sample: int) -> Dict[str, List[str]]
+        Randomly samples a specified number of images for each date.
+
+    categorize_images_into_bins(df: pd.DataFrame) -> Tuple[Dict[str, pd.DataFrame], List[str]]
+        Categorizes images into bins based on their detection counts.
+
+    sample_images_equally_from_bins(df: pd.DataFrame, percentage_ratio: float) -> pd.DataFrame
+        Samples a percentage of images equally from each bin for each date.
+
+    determine_bin_size(detection_range: int) -> int
+        Determines the bin size for categorization based on the detection range.
+    """
+    def __init__(self, input_structured_folder, customer_quality_check_folder, customer_retraining_folder, 
+                 database_parameters_json, customer_name, sampling_parameters):
+        self.input_structured_folder = input_structured_folder
+        self.customer_quality_check_folder = customer_quality_check_folder
+        self.customer_retraining_folder = customer_retraining_folder
+        self.database_parameters_json = database_parameters_json
+        self.customer_name = customer_name
+        self.sampling_parameters = sampling_parameters
+        
+    def sample_images_for_quality_check(self, grouped_images_by_date: Dict[str, List[str]] 
+    ) -> None:
+        """
+        Samples a specified number of images from each date for quality checking.
+
+        Parameters
+        ----------
+        grouped_images_by_date : Dict[str, List[str]]
+            A dictionary where keys are dates and values are lists of image file names from those dates.
+        input_structured_folder : str
+            The path of the input folder containing images.
+        customer_quality_check_folder : str
+            The destination folder path for the quality check images.
+        n_images_to_sample : int
+            Number of images to sample for each date.
+            
+        Returns
+        -------
+        None
+        """
+        
+        # Sample a number of random images for manual quality check
+        # The number is set in config.yml as quality_check_sample_size
+        n_images_to_sample = self.sampling_parameters["quality_check_sample_size"]
+        
+        logger.info(f'Sampling {n_images_to_sample} images for quality check.. \n')
+        
+        quality_check_images = SmartSampling.get_n_random_images_per_date(grouped_images_by_date, n_images_to_sample)
```

Comment:
If a method is static you need to add the decorator and remove the self.
If the method is not static then it should be called by the same object, so self.method().
Otherwise you are calling it as a class method.